sphinx/environment/__init__.py

```diff
@@ -725,7 +725,7 @@ def check_consistency(self) -> None:
                 if 'orphan' in self.metadata[docname]:
                     continue
                 logger.warning(__("document isn't included in any toctree"),
-                               location=docname)
+                               location=docname, type='toc', subtype='not_included')
```

Comment:
I would prefer `orphan` but I don't know if it could confuse people (like the warning is to warn that a document is orphan but it wasn't mark as such).

Reply:
Maybe we can change the warning log to `"document isn't included in any toctree, yet not marked as orphan in metadata."`

Reply:
I prefer a shorter message. I think the subtype could still be `orphan`, but then,  is there any possibility that `toc.orphan` could mean something else for users? I would say "ok that means there was an issue with an orphan doc, be it a doc marked explicitly with `:orphan:` or a doc detected as orphan but not marked as such". Would it make sense or would it ne too confusing? also, we could document each type/subtype and its meaning.

Otherwise we can keep `not_included`.